### DISCUSSION TITLE
Adding a new Pinia store to an existing vite project

### DISCUSSION THREAD

**Discussion by anli-xsigns** [quasarframework/quasar#13149] [General - Components / Directives / etc] [answered]
Hello,

I migrated my existing webpack project to vite using this (very good!) migration guide: https://quasar.dev/quasar-cli-vite/convert-to-quasar-cli-with-vite
However when I set up the new project I might not have chosen pinia as store. When I try to add a store via `quasar new store tests
tore --format ts` a new vuex store is created. How can I change the store to pinia? When I copy the content of the .quasar/app.js of a project with pinia everything works as expected - however the file is overwritten on next start.

Thanks in advance for your help!

**Comment by TobyMosque** [comment, marked as answer]
probably, u'd vuex installed, so the first step is check if that is installed, so remove it.

```
yarn remove vuex
```

so, install pinia:

```
yarn add pinia
```

if your store is inside `src/store/index.js` move it to `src/stores/index.js`, and make sure that make use of the `quasar/wrappers/store`:

```js
import { store } from 'quasar/wrappers'
import { createPinia } from 'pinia'

export default store (/*{ ssrContext }*/) => {
  const pinia = createPinia()
  // pinia.use(InstallPlugins)
  return pinia 
})
```

as a last resource, try to reinstall your packages:

```
npm i -g yarn@latest
rm yarn.lock
rm package-lock.json
rm -r -f node_modules
yarn
```

**Comment by CrawfordW** [comment]
Is it not possible to have both vuex and pinia in the same project? I have a large app with many vuex modules, and I want to migrate to pinia one module at a time. Can't figure out how to do this.
